### PR TITLE
Support dynamic type

### LINKED ISSUE #321: Support `dynamic` for Kotlin JS
There doesn't seem to be any direct support for the `dyanmic` type for Kotlin JS. You can hack it together using something like:

```Kotlin
val DYNAMIC = ClassName("", "dynamic")
```

It would be nice if kotlinpoet provided support for it out of the box though.

### PR DESCRIPTION
**Pull request by Egorand**
Closes #321

### CODE REVIEW

File: src/main/java/com/squareup/kotlinpoet/TypeName.kt [hunk 1]
```diff
@@ -216,6 +216,7 @@ abstract class TypeName internal constructor(
 @JvmField val CHAR = ClassName("kotlin", "Char")
 @JvmField val FLOAT = ClassName("kotlin", "Float")
 @JvmField val DOUBLE = ClassName("kotlin", "Double")
+@JvmField val DYNAMIC = ClassName("", "dynamic")
```
Comment (JakeWharton) [inline src/main/java/com/squareup/kotlinpoet/TypeName.kt]: I don't think it makes sense to model this as a `ClassName`. None of the methods on `ClassName` are relevant for it.

Comment (Egorand) [reply src/main/java/com/squareup/kotlinpoet/TypeName.kt]: Agree. What about `object Dynamic : TypeName()`?

Comment (JakeWharton) [reply src/main/java/com/squareup/kotlinpoet/TypeName.kt]: Works for me.

**Comment by NightlyNexus** [comment]
`Dynamic.INSTANCE` is a little bit unfortunate for Java consumers, but I don't have a better idea.

**Comment by Egorand** [comment]
You're right. I also thought about making `Dynamic` an anonymous class inside `TypeName` with `@JvmField`, would that be a better idea in terms of Java interop?

**Comment by NightlyNexus** [comment]
I would lean towards not sacrificing idiomatic Kotlin for better Java interop.
+1 as is.

File: src/main/java/com/squareup/kotlinpoet/Dynamic.kt [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.kotlinpoet
+
+/**
+ * JavaScript-specific `dynamic` type.
+ * See [https://kotlinlang.org/docs/reference/dynamic-type.html].
+ */
+object Dynamic : TypeName(false, emptyList()) {
```
Comment (swankjesse) [inline src/main/java/com/squareup/kotlinpoet/Dynamic.kt]: Move to TypeName.kt

**Pull request merged**